frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -210,13 +210,18 @@ function DatasetEditor(props) {
     if (!Array.isArray(orderedColumns)) {
       return columns;
     }
-    return orderedColumns
-      .map(
-        col =>
-          columns.find(c => isSameField(c.field_ref, col.fieldRef)) ||
-          virtualCardColumns.find(c => isSameField(c.field_ref, col.fieldRef)),
-      )
-      .filter(Boolean);
+
+    function getFieldFromColumnVizSetting(columnVizSetting) {
+      const fieldRef = columnVizSetting.fieldRef || columnVizSetting.field_ref;
```

Comment:
Should we maybe extract this line into a function and put it into `metabase-lib`?

Reply:
I think we should also add a comment explaining why this is needed